velox/common/base/Semaphore.h

```diff
@@ -63,8 +63,8 @@ class Semaphore {
  private:
   std::mutex mutex_;
   std::condition_variable cv_;
-  volatile int32_t count_;
-  volatile int32_t numWaiting_{0};
+  std::atomic<int32_t> count_;
```

Comment:
You can just remove the `volatile`, these are already fully protected by the mutex.

Reply:
@majetideepak was concerned that without volatile some code might be optimized away that existed previously. There must have been some reason to use it before. So I changed it to use atomic to prevent any issue as this is the recommended change for situations like this.

Reply:
With modern compiler and CPU `volatile` is not needed.  The case I would need it in my career is in CUDA.

Reply:
I defer to you on this. I have never used it either that I remember. I assume there was some reason the original author used it. Given we do have mutex protection single threaded access is guaranteed. I'm removing the std::atomic use here.